packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -402,6 +423,15 @@ function ComboBoxTray(props: ComboBoxTrayProps) {
     lastInputValue.current = inputValue;
   }, [loadingState, inputValue, showLoading]);
 
+  let onBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    // Ignore blur if focus moves from tray input to something outside the popover (i.e. tray is being closed via tapping outside the tray -> focus will return to combobox button -> prevent menuTrigger = focus from reopening the combobox tray).
+    if (e.target === inputRef?.current && !popoverRef.current?.contains(e.relatedTarget as HTMLElement)) {
+      return;
+    }
+
+    inputProps.onBlur(e);
+  };
```

Comment:
Fixes the following bug: `Can't close "menuTrigger='focus'" combobox on mobile unless you select an option`

Early returns `onBlur` so `isFocused` doesn't get set to false when you close the tray. Put it here because this feels specific to our mobile combobox implementation